lib/api/model/model.dart

```diff
@@ -682,6 +695,29 @@ extension type const TopicName(String _value) {
   /// The key to use for "same topic as" comparisons.
   String canonicalize() => apiName.toLowerCase();
 
+  /// A [TopicName] with [resolvedTopicPrefixRegexp] stripped if present.
+  ///
+  /// If [resolvedTopicPrefixRegexp] is absent, [this].
+  TopicName unresolve() {
+    final match = resolvedTopicPrefixRegexp.firstMatch(_value);
+    if (match == null) {
+      return this;
+    } else {
+      return TopicName(_value.substring(match.end));
+    }
+  }
+
+  /// Whether [this] and [other] have the same canonical form,
+  /// using [canonicalize].
+  ///
+  /// If [ignoringResolvePrefix] is true,
+  /// the [unresolve]d forms of [this] and [other] are compared instead.
+  bool isSameAs(TopicName other, {required bool ignoringResolvePrefix}) {
```

Comment:
This feels to me more like two different methods than one method with a boolean flag.

And then I think the "ignoring" case is basically doing much the same job as its one caller `topicMoveWasResolveOrUnresolve`, so can be inlined into there.